src/lib/offchain.ts

```diff
@@ -257,3 +268,134 @@ export async function fetchOffchainTokenData(
   }
   return offchainTokenData
 }
+
+// ----- Logic Replication using a chain agnostic input -----
+export type FetchOffchainTokenDataInput = {
```

Comment:
New type for the `fetchOffchainTokenData` function that abstracts events part in a common type, [ChainEvent](https://github.com/smartcontractkit/ccip-tools-ts/pull/22/files#diff-a165b00264fe1582eba3cb59a6f93c8417e9f2a1cb952d30512d723617da4c19R2), that each chain specific event can adapt to